src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3379,7 +3383,14 @@ StartPlacementExecutionOnSession(TaskPlacementExecution *placementExecution,
 	}
 	else
 	{
-		querySent = SendRemoteCommand(connection, queryString);
+		if (!strchr(queryString, ';'))
```

Comment:
That's a super unreliable check. As of https://github.com/citusdata/citus/pull/3871 you can use the queryCount in the Task.

Reply:
Thanks, I already hoped/expected there was a better way to check this. This was just a hack to get a working prototype.

Reply:
Turns out I did not even have to check the queryCount. Since when there's more than 1 query we don't have a tuple descriptor, so we don't care about results being binary or not.